### PR TITLE
fix: Handle AWS public urls separately when extracting TLDs

### PR DESCRIPTION
**Pull request by KShivendu**
## Summary of change

Handle AWS public url separately when extracting TLDs

## Related issues

-   https://github.com/supertokens/supertokens-python/issues/394
## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [x] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
-   [ ] Make sure that `syncio` / `asyncio` functions are consistent.
-   [ ] If access token structure has changed
    -   Modified test in `tests/sessions/test_access_token_version.py` to account for any new claims that are optional or omitted by the core

### CODE REVIEW

File: supertokens_python/utils.py [hunk 1]
```diff
@@ -299,8 +299,16 @@ def get_top_level_domain_for_same_site_resolution(url: str) -> str:
...
         return "localhost"
+
     parsed_url: Any = extract(hostname, include_psl_private_domains=True)
     if parsed_url.domain == "":  # type: ignore
+        if hostname.endswith(".amazonaws.com"):
+            # Example: url http://ec2-xx-yyy-zzz-0.compute-1.amazonaws.com
+            # return ec2-xx-yyy-zzz-0.compute-1.amazonaws.com
+            # If user deploys the website on the same ec2 instance as the API
+            # this use SameSite=lax, otherwise SameSite=none
+            return hostname
```
Comment (rishabhpoddar) [inline supertokens_python/utils.py:310]: - return just amazonaws.com
- modify the place where you are checking against the public suffix list to check for this domain too
- add tests

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/utils.py (hunk 1 above)
Comment (KShivendu) [reply supertokens_python/utils.py]: done

File: supertokens_python/utils.py [hunk 2]
```diff
@@ -299,8 +299,13 @@ def get_top_level_domain_for_same_site_resolution(url: str) -> str:
 
     if hostname.startswith("localhost") or is_an_ip_address(hostname):
         return "localhost"
+
     parsed_url: Any = extract(hostname, include_psl_private_domains=True)
     if parsed_url.domain == "":  # type: ignore
+        if hostname.endswith(".amazonaws.com"):
+            # tldextract.extract() isn't able to parse AWS service public urls
+            return "amazonaws.com"
+
```
Comment (rishabhpoddar) [inline supertokens_python/utils.py:308]: undo this and upgrade the lib

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by KShivendu** [comment]
Closing this in favour of https://github.com/supertokens/supertokens-python/pull/449.

**Pull request closed without merging**